contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -780,19 +831,27 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
     /*
      * Updates owners, but do not emit Transfer event. Event was already emited during pre-mint.
      */
-    function silentMintAndSetPremintStatus(
-        address _from,
-        uint256 _tokenId,
-        uint256 _offerId
-    ) internal {
-        require(_from == owner(), NO_SILENT_MINT_ALLOWED);
+    function silentMint(address _from, uint256 _tokenId) internal {
+        require(_from == owner() || _from == address(this), NO_SILENT_MINT_ALLOWED);
 
         // update data, so transfer will succeed
         getERC721UpgradeableStorage()._owners[_tokenId] = _from;
+    }
+
+    function _isApprovedOrOwner(address spender, uint256 tokenId) internal view override returns (bool) {
+        address owner = ownerOf(tokenId);
+        return (spender == owner || isApprovedForAll(owner, spender) || getApproved(tokenId) == spender);
+    }
+
+    /**
+     * @dev Reverts if the `_tokenId` has not been minted yet and is not a pre-minted token.
+     */
+    //@TODO check this
```

Comment:
I believe this is not really needed, since `silentMint` should handle it for preminted tokens (i.e. their "exists" status is changed before `requireMinted` is invoked.